server/routerlicious/packages/lambdas/src/scribe/lambda.ts

```diff
@@ -598,15 +598,28 @@ export class ScribeLambda implements IPartitionLambda {
 			this.pendingCheckpointOffset = queuedMessage;
 			return;
 		}
-
+		let databaseCheckpointFailed = false;
 		this.pendingP = clearCache
 			? this.checkpointManager.delete(this.protocolHead, true)
-			: this.writeCheckpoint(checkpoint);
+			: this.writeCheckpoint(checkpoint).catch((error) => {
+					databaseCheckpointFailed = true;
+					Lumberjack.error(
+						`Error writing database checkpoint.`,
+						getLumberBaseProperties(this.documentId, this.tenantId),
+						error,
+					);
+			  });
 		this.pendingP
 			.then(() => {
 				this.pendingP = undefined;
-				if (!skipKafkaCheckpoint) {
+				if (!skipKafkaCheckpoint && !databaseCheckpointFailed) {
 					this.context.checkpoint(queuedMessage, this.restartOnCheckpointFailure);
+				} else if (databaseCheckpointFailed) {
+					Lumberjack.info(
+						`Skipping kafka checkpoint due to database checkpoint failure.`,
```

Comment:
I think here need to be due to either database checkpoint faliure or confiiggred tot skip kafka?